paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -301,29 +285,30 @@ internal class PaymentSheetViewModel @Inject internal constructor(
         }
     }
 
-    suspend fun confirmStripeIntent(
-        authActivityStarterHost: AuthActivityStarterHost,
-        confirmStripeIntentParams: ConfirmStripeIntentParams
-    ) {
-        paymentController.startConfirmAndAuth(
-            authActivityStarterHost,
-            confirmStripeIntentParams,
-            ApiRequest.Options(
-                lazyPaymentConfig.get().publishableKey,
-                lazyPaymentConfig.get().stripeAccountId
-            )
-        )
+    fun confirmStripeIntent(confirmStripeIntentParams: ConfirmStripeIntentParams) {
+        when (confirmStripeIntentParams) {
+            is ConfirmPaymentIntentParams -> {
+                paymentLauncher?.confirm(confirmStripeIntentParams)
```

Comment:
paymentLauncher being null is an incorrect state, probably do 
```
runCatching {
    requireNotNull(paymentLauncher)
}.fold(
    onSuccess = {
        when (confirmStripeIntentParams) {
            is ConfirmPaymentIntentParams -> {
                it.confirm(confirmStripeIntentParams)
            }
            is ConfirmSetupIntentParams -> {
                it.confirm(confirmStripeIntentParams)
            }
        }
    },
    onFailure = ::onFatal
)
```

Reply:
nice catch. That might save us some headache later.

fixed in [1703bef](https://github.com/stripe/stripe-android/pull/4159/commits/1703bef6d382a02819bebae65cd66be7e6658e81)